### ISSUE TITLE
Use scala-java-time for java.time._ in scala.js distribution by default

### ISSUE THREAD

**Issue by mdedetrich** [scala-js/scala-js#2973]
Although there is currently an implementation for java.time._, its quite rudimentary and only provides basic functionality.

There currently exists a more robust implementation here https://github.com/cquiroz/scala-java-time . It was originally made by @soc , but now @cquiroz maintains it. We are currently using this version in production and have zero problems aside and its quite a full featured implementation of java.time._

Would it be possible to use this implementation and resolve any licensing issues (see https://github.com/cquiroz/scala-java-time/blob/master/LICENSE.txt)?

**Comment by sjrd** [comment]
What do you mean by "use this implementation"? Obviously you're using it, so it looks fine.

**Comment by mdedetrich** [comment]
I meant including it in the official scala.js distribution, will rename title

**Comment by sjrd** [comment]
We don't plan to include any `java.time` implementation in the "official" Scala.js distribution. It is meant to stay in a separate repository. There's no reason to put it in the core repo. It would only burden us with a bigger repo and longer CI times.

**Comment by mdedetrich** [comment]
Even if its maintained by the original official author and its idiomatic Scala code with tests? The main reason behind the ticket is that, a standard date time library is something that was sorely missing when working with Scala.js, and its basically something that I find needing all the time.

In the end, I suppose this comes down to what is "required" and what is not (similar with scala-platform discussion). For example, I would argue that `java.nio._` is less important than `java.time._`, I have very rarely, if ever, myself (or seen other people) use whats inside `nio`

**Comment by sjrd** [comment]
> I suppose this comes down to what is "required" and what is not

In our case, that's very well defined. Something is "required" if we either

* It requires knowledge of the internals, help from the compiler, or is otherwise intrinsic, or
* We need it ourselves to test other parts of our own build

For example:
* JUnit is required because we need it to test our compiler and library (obviously). This means we have JUnit in the core repo, even though one might think it deserves a completely different repository.
* `java.math.BigInteger` and `BigDecimal` are required because they are touched by some parts of the Scala library, which we need to test
* `java.nio._` is required because it requires some deep interaction with Typed Arrays, which can only be done if they're in the same repo

But:
* PhantomJS support is not required, and has moved out in 1.x
* `jsDependencies` which represents half of the sbt plugin in 0.6.x is being moved out in 1.x, because it is not needed for our own purposes (or at least, won't be anymore once #2974 is merged).
* `java.time._` is not required by anything else in the core repository, and can be compiled and tested separately
* `scalajs-dom` is not required either

**Comment by sjrd** [comment]
Closing as I think I have answered the question. Feel free to reopen if necessary.
